Embedding Sandbox docs (#121)
apollographql/docs: src/content/studio/explorer/sandbox.mdx

In addition to using [Studio's hosted version of Sandbox](https://studio.apollographql.com/sandbox), you can also embed Sandbox on your own website.
This may be useful for bootstrapping a GraphQL IDE onto a development endpoint or an endpoint with CORS restrictions.

### Setup

1. <a href="https://studio.apollographql.com/sandbox" target="_blank">
     Go to Sandbox
   </a>

> 2. Under the Explorer's Settings tab, find **Embed Sandbox** and click **Copy code snippet**.

Reviewer: @mayakoneval Do we also have the option on Editor pane? like we do for Explorer?

Author: We have the option but it doesn't do anything different. In Explorer it embed your current tab, but in Sandbox we don't give you the option to embed a document by default. I can add it as well 👍

   The following dialog appears:

   <img
     class="screenshot"
     src="../img/embed-sandbox-modal.jpg"
     alt="Sandbox embed dialog"
     width="600"
   />

3. Click **Copy** to copy the snippet, then paste it into your code.

### Options

The `EmbeddedSandbox` object takes an options object with the following structure (individual options are described below):

```js
{
  initialEndpoint: 'http://localhost:4000',
  handleRequest: (endpointUrl, options) => {
    return fetch(endpointUrl, {
      ...options,
      headers: {
          ...options.headers,
          authorization: `token ${token}`
      },
    })
  },
}
```

These are the optional options you can include in the object you pass to `new EmbeddedSandbox`:

<table class="field-table api-ref">

<thead>
  <tr>
    <th>
      Name /<br />
      Type
    </th>
    <th>Description</th>
  </tr>
</thead>

<tbody>
<tr>
<td>

##### `initialEndpoint`

`string`

</td>
<td>

> This will be the initial endpoint introspected wjen the Sandbox renders on page load. 

Reviewer: ```suggestion
This will be the initial endpoint introspected when the Sandbox renders on page load. 
```

> This will be the initial endpoint introspected when the Sandbox renders on page load. 

Reviewer 2: ```suggestion
The URL of the GraphQL endpoint that Sandbox introspects on initial load. Sandbox populates its pages using the schema obtained from this endpoint.

The default value is `http://localhost:4000`.
```

> The schema at this endpoint will be used to populate the Sandbox pages. Defaults to `http://localhost:4000`.

Reviewer 2: ```suggestion
```

> In general, we recommend you **only pass development endpoints to Sandbox**.

Reviewer 2: ```suggestion
**You should only pass non-production endpoints to Sandbox**. Sandbox is powered by schema introspection, and we recommend [disabling introspection in production](https://www.apollographql.com/blog/graphql/security/why-you-should-disable-graphql-introspection-in-production/).
```

> Sandbox is powered by schema introspection and we recommend [turning off introspection in production](https://www.apollographql.com/blog/graphql/security/why-you-should-disable-graphql-introspection-in-production/).

Reviewer 2: ```suggestion
```

> If you want to provide a **similar experience to embedded Sandbox on a production endpoint**, we recommend using Studio's [Public Graphs](https://www.apollographql.com/docs/studio/org/graphs#public-variants) or [embedded Explorer](https://www.apollographql.com/docs/studio/explorer/embed-explorer).

Reviewer 2: ```suggestion
To provide a "Sandbox-like" experience for production endpoints, we recommend using either a [public variant](https://www.apollographql.com/docs/studio/org/graphs#public-variants) or the [embedded Explorer](https://www.apollographql.com/docs/studio/explorer/embed-explorer).
```

</td>
</tr>

<tr>
<td>

##### `handleRequest`

`(url, options) => Promise`

</td>
<td>

> By default, the embedded Sandbox uses the `fetch` API to send requests to the endpoint url at the top of the page from your webpage. You can instead pass a _custom_ fetcher that's used to run all operations.

Reviewer 2: ```suggestion
By default, the embedded Sandbox uses the `fetch` API to send requests from your webpage to your specified GraphQL endpoint. If you provide a custom `handleRequest` function, Sandbox uses it instead of `fetch` for all operations.
```

You might want to do this if you need to include specific headers in every request made from your embedded Sandbox.

</td>
</tr>

<tr>
<td>

##### `includeCookies`

`boolean`

</td>
<td>

By default, the embedded Sandbox uses the default `fetch` function to make requests, which passes `{ credentials: 'omit' }`.

> You can pass `includeCookies: true` if you want `{ credentials: 'include' }` to be passed as an option whenever requests are made from Explorer. If you pass a `handleRequest`, this option will be ignored.

Reviewer 2: ```suggestion
You can set `includeCookies` to `true` if you instead want the Explorer to pass `{ credentials: 'include' }` for its requests.

If you pass the `handleRequest` option, this option is ignored.
```

> You can pass `includeCookies: true` if you want `{ credentials: 'include' }` to be passed as an option whenever requests are make from the Explorer. If you pass a `handleRequest`, this option will be ignored.

Reviewer: ```suggestion
You can pass `includeCookies: true` if you want `{ credentials: 'include' }` to be passed as an option whenever requests are made from Explorer. If you pass a `handleRequest`, this option will be ignored.
```

In addition to using [Studio's hosted version of Sandbox](https://studio.apollographql.com/sandbox), you can also embed Sandbox on your own website.
This may be useful for bootstrapping a GraphQL IDE onto a development endpoint or an endpoint with CORS restrictions.

### Setup

1. <a href="https://studio.apollographql.com/sandbox" target="_blank">
     Open Sandbox.
   </a>

2. Under the Explorer's Settings tab, find **Embed Sandbox** and click **Copy code snippet**.

   The following dialog appears:

   <img
     class="screenshot"
     src="../img/embed-sandbox-modal.jpg"
     alt="Sandbox embed dialog"
     width="600"
   />

> 3. Click **Copy** to copy the snippet, then paste it into your code.

Author: ```suggestion
3. Click **Copy** to copy the snippet, then paste it into your html code.
```
